File: Frontend/src/interfaces/v12/size/size.ts
import { CreatedChangedEntity } from "interfaces/v12/CreatedChangedEntity";
import { guid } from "types/guid";
import { PagedRequestAbstractDto } from "../pagedRequest/pagedRequestAbstractDto";

export interface SizeDetailsDto extends CreatedChangedEntity {
  id: guid;
  name: string;
}

export interface SizeCreateUpdateDto {
  name: string;
}

export interface SizeSearchDto extends PagedRequestAbstractDto {
  search: string;
}

File: Wms.Api/Dto/Size/Size.cs
using Wms.Api.Entities; 

namespace Wms.Api.Dto.Size
{
    public class SizeDetailsDto : CreatedChangedEntity
    {
        public Guid Id { get; set; }
        public required string Name { get; set; }
    }

    public class SizeCreateUpdateDto
    {
        public required string Name { get; set; }
    }

    public class SizeSearchDto : PagedRequestAbstractDto
    {
        public required string Search { get; set; }
    }
}
